File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.go
func check(nums []int) bool {
    n := len(nums)
    count := 0
    
    for i := 0; i < n; i++ {
        if nums[i] > nums[(i+1)%n] {
            count++
        }
    }
    
    // The array is sorted and rotated if there is at most one point where nums[i] > nums[i+1]
    return count <= 1
}

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.rs
impl Solution {
    pub fn check(nums: Vec<i32>) -> bool {
        let mut count = 0;
        let n = nums.len();

        for i in 1..n {
            if nums[i-1] > nums[i] {
                count += 1;
            }
        }

        if nums[n-1] > nums[0] {
            count += 1;
        }

        count <= 1
    }
}
